src/transformState.js

```diff
@@ -4,8 +4,30 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
```

Comment:
this can also be achieved using [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) which is more common nowadays

Reply:
@choeqq In this particular example I would have to reassign value to the state
state = { ...state, ...action.extraData };
which I cannot do with function parameters, so Object.assign is the best option here.